subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/FileSystemSnapshotBuilder.java

```diff
@@ -53,7 +54,7 @@ public void addDir(File dir, String[] segments) {
     public void addFile(File file, String[] segments, String fileName, FileMetadataSnapshot metadataSnapshot) {
         checkNoRootFileSnapshot("another root file", file);
         HashCode contentHash = fileHasher.hash(file, metadataSnapshot.getLength(), metadataSnapshot.getLastModified());
-        RegularFileSnapshot fileSnapshot = new RegularFileSnapshot(stringInterner.intern(file.getAbsolutePath()), fileName, contentHash, FileMetadata.from(metadataSnapshot));
+        RegularFileSnapshot fileSnapshot = new RegularFileSnapshot(stringInterner.intern(file.getAbsolutePath()), fileName, contentHash, FileMetadata.from(metadataSnapshot), AccessType.DIRECT);
```

Comment:
Why is this always `DIRECT`? Can you explain in a comment?

Reply:
The latest code on this branch does not use `DIRECT` all the time.